File: 0278-first-bad-version/0278-first-bad-version.py
# The isBadVersion API is already defined for you.
# def isBadVersion(version: int) -> bool:

class Solution:
    def firstBadVersion(self, n: int) -> int:
        L, R = 1, n

        while L <= R:
            mid = L + (R - L) // 2

            if isBadVersion(mid):
                R = mid - 1
            else:
                L = mid + 1

        return L

File: 0278-first-bad-version/0278-first-bad-version.cpp
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        int L = 1, R = n;

        while (L <= R)
        {
            int mid = L + (R - L)/2;
            if (isBadVersion(mid))
                R = mid - 1;
            else
                L = mid + 1;
        }
        return L;
    }
};